libs/fetcher.client.js

```diff
@@ -382,13 +345,8 @@ Queue.prototype = {
             use_post = request.operation !== OP_READ || config.post_for_read;
 
             if (!use_post) {
-                if (lodash.isFunction(config.constructGetUri)) {
-                    get_uri = config.constructGetUri.call(this, uri, request.resource, request.params, config);
-                }
-
-                if (!get_uri) {
-                    get_uri = this._defaultConstructGetUri(uri, request.resource, request.params, config);
-                }
+                var getUriFn = lodash.isFunction(config.constructGetUri) ? config.constructGetUri : defaultConstructGetUri;
+                var get_uri = getUriFn.call(this, uri, request.resource, request.params, config, this.context);
```

Comment:
I did add one more param `this.context` to the construct get uri function, since that is most likely the only thing the function needs from `this` object (at least for the default `constructGetUri` function).  Should not break compatibility.